server/database/models/User.js

```diff
@@ -0,0 +1,35 @@
+const mongoose = require('mongoose');
+const bcrypt = require('bcrypt');
+
+const UserSchema = new mongoose.SchemaType({
+  email: {
+    type: String,
+    default: '',
+  },
+  password: {
+    type: String,
+    default: '',
+  },
+  name: {
+    type: String,
+    default: '',
+  },
+  isDeleted: {
+    type: Boolean,
+    default: false,
+  },
+  signUpDate: {
+    type: Date,
+    default: Date.now(),
+  },
+});
+
+UserSchema.methods.generateHash = password => {
+  return bcrypt.hashSync(password, bcrypt.getSaltSync(8), null);
+};
+
+UserSchema.methods.validPassword = password => {
```

Comment:
sorry @teenie-quaggard  so many Q !
but can I have a ref for how this method works?

Reply:
https://mongoosejs.com/docs/2.7.x/docs/methods-statics.html

this is basically just me creating a function that checks if a password is valid!